conans/test/functional/toolchains/cmake/test_file_api.py

```diff
@@ -0,0 +1,142 @@
+import os
+import textwrap
+
+import pytest
+
+from conans.test.assets.cmake import gen_cmakelists
+from conans.test.assets.sources import gen_function_cpp, gen_function_h
+from conans.test.utils.tools import TestClient
+
+
+@pytest.mark.tool_cmake
+def test_file_api():
+    """
+    simple library providing 3 targets:
+    - decoder
+    - encoder
+    - transcoder (requires decoder and encoder)
+    """
+    client = TestClient()
+
+    conanfile = textwrap.dedent("""
+        from conans import ConanFile
+        from conan.tools.cmake import CMake, CMakeFileAPI
+
+        class Triunfo(ConanFile):
+            name = "triunfo"
+            version = "1.0"
+            settings = "os", "compiler", "arch", "build_type"
+            exports_sources = "*CMakeLists.txt", "*.cpp", "*.h"
+            generators = "CMakeToolchain"
+            _cmake = None
+
+            def _configure_cmake(self):
```

Comment:
This idiom should not be longer necessary with CMakeDeps and CMakeToolchain, please avoid using this ``_configure_cmake()`` method.

Reply:
how come it is no longer necessary? it will invoke `configure` twice, once for `build`, and once `package` otherwise

Reply:
package() is called after build(), no need to call configure() again.
That was due to an old misbehavior in the helper that required it to be called twice to do the right thing. No longer necessary with the new toolchain.